internal/storage/bucket_handle.go

```diff
@@ -96,3 +96,40 @@ func (b *bucketHandle) StatObject(ctx context.Context, req *gcs.StatObjectReques
 
 	return
 }
+
+func (bh *bucketHandle) CreateObject(ctx context.Context, req *gcs.CreateObjectRequest) (o *gcs.Object, err error) {
+	obj := bh.bucket.Object(req.Name)
+
+	if req.GenerationPrecondition != nil && *req.GenerationPrecondition != 0 {
+		if req.MetaGenerationPrecondition != nil && *req.MetaGenerationPrecondition != 0 {
```

Comment:
can you add a comment what does generation and metageneration preconditions do in case of createObject